cloudsync/event.py

```diff
@@ -193,22 +193,19 @@ def do(self):
             self.backoff()
 
     def _do_walk_if_needed(self):
-        if self.need_walk:
+        if self.need_walk and self._root_oid:
```

Comment:
nit: a separate check for `_root_oid` ahead of this check, that logs the condition and returns early, could help with the logging of this condition

Reply:
thought about it, don't see a lot of benefit to logging that in the do loop, where we would need to also add a check to avoid excessive logging.

operating without a root is sort of supported, and we do log the root in other places.